app/src/main/kotlin/github/com/raccok/dota2androidapp/MainActivity.kt

```diff
@@ -93,12 +104,94 @@ class MainActivity : Activity() {
     }
   }
 
+  private fun queryHeroNames() {
+    // Check if the device is connected to the internet
+    if (!deviceIsOnline()) {
+      Toast.makeText(applicationContext,
+                     "Querying heroes from the Dota 2 API failed (no internet connection)",
+                     Toast.LENGTH_LONG).show()
+      return
+    }
+
+    // Create a client to perform networking
+    val client = AsyncHttpClient()
+
+    // Additional parameters needed to query the Dota 2 API for currently available heroes
+    // (TODO: the API key is hard-coded temporarily, until we have Steam user authentication)
+    val params = "IEconDOTA2_570/GetHeroes/v1?key=F9FEEEDF9C7762EA5D76F2EEDB7A08BC&language=en_us"
+
+    // Have the client get a JSONObject of data and define how to respond
+    client.get(STEAM_API_URL + params, object : JsonHttpResponseHandler() {
```

Comment:
I would suggest using a framework for services. Something like Retrofit, instead of just doing http calls.
https://android.jlelse.eu/kotlin-and-retrofit-2-tutorial-with-working-codes-333a4422a890

Reply:
Agree, I would like to add something as well : try creating a POJO/model/data class to contain the response, makes it easier to use the resulting response, rather than parsing the response every time you make a http calls.

Reply:
Thanks for your suggestions, which should now be realized.